Library/Formula/gcc49.rb

```diff
@@ -80,6 +61,7 @@ def pour_bottle?
   cxxstdlib_check :skip
 
   def install
+    ENV.no_optimization
```

Comment:
Did this break in all circumstances, or did it depend on what the bootstrapping compiler was?

Reply:
This is the first time I've gone back to trying to compiler compilers on Tigerbrew with GCC 4.0.1 on Tiger.
Previous attempts have always started with 4.2.
I was able to build 4.4 & 4.5 on Leopard/PowerPC without issue using GCC 4.2 however the issue you ran into with the stage comparison is masked by using `--with-build-config=bootstrap-debug` where as it is related to optimisation.
https://github.com/Homebrew/legacy-homebrew/pull/38596
I'll be able to give a proper explanation and fix once I've done more testing.
I switched it off as 4.8 & 4.9 builds blow up late in the process and waste a very large chunk of time +  I wanted to spare my PowerBook. :)

Reply:
I don't have a fix since I can't really ruby and what I tried didn't work, but it turns out that we always build with `-Os` via the superenv & the `cc` ENV wrappers. All setting `no_optimization` really does is ensures that `-march` is not applied. (check the log file end with `.cc` for a package to see superenv performing the substitution.
It would be

Reply:
It seems all formulas from GCC 4.5 to 14 need optimisation turned off. I just did it pre-emptively on this and 4.8's formula.

Reply:
ok to land it?

Reply:
That's fine, go for it!